int32 distance
uint8 obstructed1
uint8 obstructed2
